src/Game/CapturePointSystem.cpp

```diff
@@ -0,0 +1,177 @@
+    if (m_Team1NextPossibleCapturePoint == m_NotACapturePoint && (int)capturePoint["IsHomeCapturePointForTeamNumber"] == 1) {
+        m_Team1NextPossibleCapturePoint = capturePoint["CapturePointNumber"];
+        m_Team1HomeCapturePoint = capturePoint["CapturePointNumber"];//needed to calculate next m_Team1NextPossibleCapturePoint
+    }
+    else if (m_Team2NextPossibleCapturePoint == m_NotACapturePoint && (int)capturePoint["IsHomeCapturePointForTeamNumber"] == 2) {
+        m_Team2NextPossibleCapturePoint = capturePoint["CapturePointNumber"];
+        m_Team2HomeCapturePoint = capturePoint["CapturePointNumber"];//needed to calculate next m_Team2NextPossibleCapturePoint
+    }
+    //at least one capturepoint has been taken over
+    //do nothing, its being handled inside the next code:
+
+    //create data to be used in option B
+    //check so this is the next possible capture point for the take-over team and see if only one team is standing inside it
+    double timerDeltaChange = 0.0;
+    int currentTeam = 0;
+    if (firstTeamPlayersStandingInside > 0 && secondTeamPlayersStandingInside == 0
+        && m_Team1NextPossibleCapturePoint == (int)capturePoint["CapturePointNumber"])
+    {
+        timerDeltaChange = firstTeamPlayersStandingInside*dt;
+        currentTeam = 1;
+    }
+    else if (firstTeamPlayersStandingInside == 0 && secondTeamPlayersStandingInside > 0
+        && m_Team2NextPossibleCapturePoint == (int)capturePoint["CapturePointNumber"])
+    {
+        timerDeltaChange = -secondTeamPlayersStandingInside*dt;
+        currentTeam = 2;
+    }
+
+    //A.nobodys standing inside
+    if (firstTeamPlayersStandingInside == 0 && secondTeamPlayersStandingInside == 0) {
+        //do nothing (?)
+    }
+
+    //B. at most one of the teams have players inside (this means datavariable currentTeam is not 0)
+    else if (currentTeam != 0) {
+        //if capturePoint is not owned by the take-over team, just modify the CaptureTimer accordingly
+        if (ownedBy != currentTeam) {
+            capturePoint["CaptureTimer"] = (double)capturePoint["CaptureTimer"] + timerDeltaChange;
+        }
+        //if capturePoint is owned by the team, and the other team has been trying to take it, then increase/decrease the timer towards 0.0
+        if ((ownedBy == currentTeam && currentTeam == 1 && (double)capturePoint["CaptureTimer"] < 0.0) ||
+            (ownedBy == currentTeam && currentTeam == 2 && (double)capturePoint["CaptureTimer"] > 0.0)) {
+            capturePoint["CaptureTimer"] = (double)capturePoint["CaptureTimer"] + timerDeltaChange;
+        }
+        //check if captureTimer > m_CaptureTimeToTakeOver and if so change owner and publish the eCaptured event
+        if (abs((double)capturePoint["CaptureTimer"]) > abs(m_CaptureTimeToTakeOver)) {
+            capturePoint["OwnedBy"] = currentTeam;
+            capturePoint["CaptureTimer"] = 0.0;
```

Comment:
Set a new color on the model component when the capture point is captured would be great for debugging purposes. Testing if the next capture point etc. is set properly.